src/os/bluestore/BlueStore.cc

```diff
@@ -20558,6 +20566,20 @@ Allocator* BlueStore::clone_allocator_without_bluefs(Allocator *src_allocator)
     return nullptr;
   }
 
+  // The goal here is to obtain allocations used by Objects;
+  // we do it by subtracting allocations used by BlueFS.
+  // We need to execute:
+  // - copy_allocator
+  // - foreach_block_extents
+  // Should dispose thread be running during this,
+  // we can have src_allocator updates after we copy its contents.
+  //
+  // Hence we need:
+  // 1. Drain or cancel BlueFS dispose queue
+  // 2. copy_allocator
+  // 3. foreach_block_extents
+  bluefs->drain_discard(); // or cancel_discard() if we are in hurry
```

Comment:
Correct, BlueFS is always in RW.
RocksDB should be down; it already is when invoked from `BlueStore::store_allocator`, and in `BlueStore::push_allocation_to_rocksdb` is not, but should be fixed anyway.

BlueFS only makes actions on demand, so we do not have to worry that it will spontaneously allocate something.
Given that, I am not even looking at the cases you highlighted.